packages/ai-openai/src/node/openai-language-model.ts

```diff
@@ -185,25 +180,27 @@ export class OpenAiModel implements LanguageModel {
 export class OpenAiModelUtils {
 
     protected processSystemMessages(
-        messages: LanguageModelRequestMessage[],
+        messages: LanguageModelMessage[],
         developerMessageSettings: DeveloperMessageSettings
-    ): LanguageModelRequestMessage[] {
+    ): LanguageModelMessage[] {
         if (developerMessageSettings === 'skip') {
             return messages.filter(message => message.actor !== 'system');
         } else if (developerMessageSettings === 'mergeWithFollowingUserMessage') {
             const updated = messages.slice();
             for (let i = updated.length - 1; i >= 0; i--) {
                 if (updated[i].actor === 'system') {
+                    const systemMessage = updated[i] as TextMessage;
```

Comment:
Should we do some verification instead of blind casting?

Reply:
a system actor can only be a TextMessage. and below the messages must be of type text too as we couldn't merge otherwise. furthermore at this moment we can't have a toolresult which would be the only other alternative for a user message